### PR TITLE
Build test image on Github action

### PR DESCRIPTION
**Pull request by guilhermef**
This PR will change how tests run on Github Actions, 
ensuring a cleaner environment and removing the need 
of pushing new images when changing build dependencies.

To reduce the build time, all images are cached.

### CODE REVIEW

File: TestDockerfile38 [hunk 1]
```diff
@@ -1,14 +1,29 @@
 FROM python:3.8-slim
 WORKDIR /app
-RUN apt-get update -y && apt-get install -y python3-dev libcurl4-openssl-dev libgnutls28-dev libjpeg-progs libimage-exiftool-perl gifsicle scons python3-all-dev libboost-python-dev libexiv2-dev ffmpeg make zlib1g-dev gcc libssl-dev libjpeg-dev libwebp-dev redis && apt-get clean
-RUN apt-get update -y && apt-get install --reinstall -y build-essential && apt-get clean
-RUN pip install --upgrade pip
-COPY setup.py /app/setup.py
-RUN mkdir -p /app/thumbor
-RUN touch /app/thumbor/__init__.py
+RUN apt-get update -y && \
+    apt-get install -y \
```
Comment (marcelometal) [inline TestDockerfile38:4]: Adding `–no-install-recommends` and  `–no-install-suggests` will avoid unnecessary packages

File: TestDockerfile39 [hunk 2]
```diff
@@ -1,13 +1,28 @@
 FROM python:3.9-slim
 WORKDIR /app
-RUN apt-get update -y && apt-get install -y python3-dev libcurl4-openssl-dev libgnutls28-dev libjpeg-progs libimage-exiftool-perl gifsicle scons python3-all-dev libboost-python-dev libexiv2-dev ffmpeg make zlib1g-dev gcc libssl-dev libjpeg-dev libwebp-dev redis && apt-get clean
-RUN apt-get update -y && apt-get install --reinstall -y build-essential && apt-get clean
-RUN pip install --upgrade pip
-COPY setup.py /app/setup.py
-RUN mkdir -p /app/thumbor
-RUN touch /app/thumbor/__init__.py
+RUN apt-get update -y && \
+    apt-get install -y \
```
Comment (marcelometal) [inline TestDockerfile39:4]: Adding `–no-install-recommends` and  `–no-install-suggests` will avoid unnecessary packages

File: TestDockerfile37 [hunk 3]
```diff
@@ -1,14 +1,29 @@
 FROM python:3.7-slim
 WORKDIR /app
-RUN apt-get update -y && apt-get install -y python3-dev libcurl4-openssl-dev libgnutls28-dev libjpeg-progs libimage-exiftool-perl gifsicle scons python3-all-dev libboost-python-dev libexiv2-dev ffmpeg make zlib1g-dev gcc libssl-dev libjpeg-dev libwebp-dev redis && apt-get clean
-RUN apt-get update -y && apt-get install --reinstall -y build-essential && apt-get clean
-RUN pip install --upgrade pip
-COPY setup.py /app/setup.py
-RUN mkdir -p /app/thumbor
-RUN touch /app/thumbor/__init__.py
+RUN apt-get update -y && \
+    apt-get install -y \
```
Comment (marcelometal) [inline TestDockerfile37:4]: Adding `–no-install-recommends` and  `–no-install-suggests` will avoid unnecessary packages

File: TestDockerfile37 [hunk 4]
```diff
@@ -1,14 +1,29 @@
...
+        libexiv2-dev \
+        ffmpeg \
+        make \
+        zlib1g-dev \
+        gcc \
+        libssl-dev \
+        libjpeg-dev \
+        libwebp-dev \
+        redis && \
+    apt-get install --reinstall --no-install-recommends --no-install-suggests -y build-essential && \
```
Comment (scorphus) [inline TestDockerfile37:23]: Seems to me like this line could use a `-y`
```suggestion
    apt-get install --reinstall -y --no-install-recommends --no-install-suggests -y build-essential && \
```

File: TestDockerfile39 [hunk 5]
```diff
@@ -1,13 +1,28 @@
...
+        libexiv2-dev \
+        ffmpeg \
+        make \
+        zlib1g-dev \
+        gcc \
+        libssl-dev \
+        libjpeg-dev \
+        libwebp-dev \
+        redis && \
+    apt-get install --reinstall --no-install-recommends --no-install-suggests -y build-essential && \
```
Comment (scorphus) [inline TestDockerfile39:23]: Ditto.
```suggestion
    apt-get install --reinstall -y --no-install-recommends --no-install-suggests -y build-essential && \
```

**Review by scorphus — CHANGES_REQUESTED** [review]
Good job 👍 

Please consider my suggestions below.

File: TestDockerfile37 (hunk 4 above)
Comment (guilhermef) [reply TestDockerfile37:23]: But it has already, it's before the first package

Comment (scorphus) [reply TestDockerfile37:23]: True. I clearly missed that (along with the fact that it builds lol). That may be an indicaiton taht we're better keep those lines consistent throughout those Dockerfiles. I'll edit the suggestions.

**Review by scorphus — CHANGES_REQUESTED** [review]
Should the other Dockerfiles be patched instead? I mean, should `-y` be placed after all the other options (before the first package)? I'm happy rearranging the suggestions.

**Comment by guilhermef** [comment]
Yes, it's a good idea to keep it consistent.

**Review by scorphus — APPROVED** [review]
Nice one, @guilhermef 👍 Thanks!

**Pull request merged**